build/common.sh

```diff
@@ -490,16 +490,19 @@ function kube::build::build_image_cross() {
   local -r build_context_dir="${LOCAL_OUTPUT_ROOT}/images/${KUBE_BUILD_IMAGE}/cross"
   mkdir -p "${build_context_dir}"
   cp build/build-image/cross/Dockerfile ${build_context_dir}/Dockerfile
-  kube::build::docker_build "${KUBE_BUILD_IMAGE_CROSS}" "${build_context_dir}"
+  kube::build::docker_build "${KUBE_BUILD_IMAGE_CROSS}" "${build_context_dir}" 'true'
 }
 
 # Build a docker image from a Dockerfile.
 # $1 is the name of the image to build
 # $2 is the location of the "context" directory, with the Dockerfile at the root.
+# $3 is the value to set the --pull flag for docker build; false by default
 function kube::build::docker_build() {
   local -r image=$1
   local -r context_dir=$2
-  local -ra build_cmd=("${DOCKER[@]}" build -t "${image}" "${context_dir}")
+  local -r pull="${3:-false}"
+  local -r pull_flag="--pull=${pull}"
+  local -ra build_cmd=("${DOCKER[@]}" build -t "${image}" "${pull_flag}" "${context_dir}")
```

Comment:
The golang pull is a substantial part of the `kubernetes-build` time, when it happens, though I have no idea how `kubernetes-build` crept all the way up to 10minutes (just looking at the delta between `kubernetes-build-1.0` and `kubernetes-build-1.1` is pretty remarkably). I think I clocked it at an additional 2m, though that was a couple of `jenkins-masters` ago, so the one we're at right now may have better IOPs on the boot disk.

This doesn't often matter because we end up blocked on everything else, but for things like the PR builder, we're starting to get in really bad territory with *just the build*, it looks like, and this won't help.